src/lua/init.lua

```diff
@@ -194,15 +211,31 @@ local function search(name)
 end
 
 -- loader_preload 1
-table.insert(package.loaders, 2, cwd_loader_func(false))
-table.insert(package.loaders, 3, cwd_loader_func(true))
-table.insert(package.loaders, 4, rocks_loader_func(false))
-table.insert(package.loaders, 5, rocks_loader_func(true))
--- package.path   6
--- package.cpath  7
--- croot          8
+table.insert(package.loaders, 2, get_loader_func({
+    search_fn = get_search_func({ path_fn = fio.cwd, templates = LUA_TEMPLATES }),
+    load_fn = load_lua }))
+table.insert(package.loaders, 3, get_loader_func({
+    search_fn = get_search_func({ path_fn = fio.cwd, templates = LIB_TEMPLATES }),
+    load_fn = load_lib }))
+table.insert(package.loaders, 4, get_loader_func({
+    search_fn = get_search_func({ path_fn = approot, templates = LUA_TEMPLATES }),
+    load_fn = load_lua }))
+table.insert(package.loaders, 5, get_loader_func({
+    search_fn = get_search_func({ path_fn = approot, templates = LIB_TEMPLATES }),
+    load_fn = load_lib }))
+table.insert(package.loaders, 6, get_loader_func({
+    search_fn = get_search_func({ path_fn = approot, templates = ROCKS_LUA_TEMPLATES, traverse = true }),
+    load_fn = load_lua }))
+table.insert(package.loaders, 7, get_loader_func({
+    search_fn = get_search_func({ path_fn = approot, templates = ROCKS_LIB_TEMPLATES, traverse = true }),
```

Comment:
We try to load packages from cwd and approot both, but rocks only from approot. Don't sure whether it is right, but looks inconsistent. Say, we cannot state that setting an approot directory don't break cwd-relative requires.

Reply:
Fixed it. See comment above